contrib/cf-doc/cf-doc.sh

```diff
@@ -0,0 +1,160 @@
+#!/bin/bash
+# Author: Mike Weilgart
+# Date: 18 January 2019
+# Purpose: Extract summary documentation from specially prepared CFEngine policy.
+
+# url_prefix should have a trailing slash
+url_prefix=https://example.com/cfengine/masterfiles/blob/
+
+collection='
+{
+  "metapromisers": {
+    "inventory": "Inventory",
+    "config": "Configuration"
+  },
+  "metatags": {
+    "docinv": "Inventory",
+    "docconfig": "Configuration"
+  }
+}
+'
+
+########################################################################
+#                      USAGE MESSAGE                                   #
+########################################################################
+usage() {
+cat >&2 <<EOF
+SYNOPSIS
+        cf-doc --help
+        cf-doc [-f FILE] [-p VERSION] [-u URL] [-t]
+
+DESCRIPTION
+        cf-doc will extract particular meta tags and meta promises
+        from CFEngine policy and will output a list of them in
+        Markdown format, to document and index a codebase of
+        CFEngine policy.
+
+        Each line of output is formatted as a URL linking back
+        to the line of policy from which it was extracted.
+        The -t option can also be used to generate plain text
+        (no hyperlinks).
+
+        cf-promises is used internally to parse CFEngine policy and
+        generate json output, which is further parsed to produce
+        the final script output.
+
+        The particular meta tags and meta promises to be collected
+        are as defined in the "collection" variable in this script.
+        For further details on this see the inline documentation
+        in the supporting script 'extract-cf-meta.jq'.
+
+OPTIONS
+        -f FILE         File to pass to cf-promises for parsing.
+                        If not specified cf-promises will use its own
+                        default of /var/cfengine/inputs/promises.cf
+
+        -p VERSION      Version of the policy that is being checked,
```

Comment:
The version of the MPF policy? Perhaps it could try to derive the default from the MPF directly.
https://github.com/cfengine/masterfiles/blob/master/promises.cf.in#L52
But, I don't think there is yet a variable that exposes this. I think I would call it sys.policy_entry_version if we added one.

Reply:
Really the purpose is to allow linking directly to the line of code in GitLab or GitHab or other version control system.  Just exactly like the link you just wrote.  :)  It contains "master" and hence the default value of master.